Sources/Moya/TargetType.swift

```diff
@@ -35,32 +29,33 @@ public extension TargetType {
     }
 }
 
-/// Represents a type of upload task.
-public enum UploadType {
+/// Represents an HTTP task.
+public enum Task {
 
-    /// Upload a file.
-    case file(URL)
+    /// A request with no additional data.
+    case requestPlain
 
-    /// Upload "multipart/form-data"
-    case multipart([MultipartFormData])
-}
+    /// A requests body set with data.
+    case requestData(Data)
 
-/// Represents a type of download task.
-public enum DownloadType {
+    /// A requests body set with encoded parameters.
+    case requestParameters(parameters: [String: Any], encoding: ParameterEncoding)
 
-    /// Download a file to a destination.
-    case request(DownloadDestination)
-}
+    /// A requests body set with data, combined with url parameters.
+    case requestCompositeData(urlParameters: [String: Any], bodyData: Data)
```

Comment:
I think it would be better to swap the order of the `urlParameters` and `bodyData` associated values. The first associated value should match the last word of the enum case.